pkg/doc/jose/decrypter.go

```diff
@@ -137,29 +138,33 @@ func (jd *JWEDecrypt) addSenderKey(skid string) error {
 	return nil
 }
 
-func (jd *JWEDecrypt) validateAndExtractProtectedHeaders(jwe *JSONWebEncryption) (Headers, string, error) {
+func (jd *JWEDecrypt) validateAndExtractProtectedHeaders(jwe *JSONWebEncryption) (Headers, string, string, error) {
 	if jwe == nil {
-		return nil, "", fmt.Errorf("jwe is nil")
+		return nil, "", "", fmt.Errorf("jwe is nil")
 	}
 
 	protectedHeaders := jwe.ProtectedHeaders
 
 	encAlg, ok := protectedHeaders.Encryption()
 	if !ok {
-		return nil, "", fmt.Errorf("jwe is missing encryption algorithm 'enc' header")
+		return nil, "", "", fmt.Errorf("jwe is missing encryption algorithm 'enc' header")
 	}
 
+	// TODO go Jose doesn't set/enforce content `typ` in the protected headers when encrypting. When fixed, remove
+	//      the following line and add a check to ensure it's available.
+	encType := "didcomm-envelope-enc" // used by authcrypt/anoncrypt. The jose package is not used for LegacyPacker.
```

Comment:
there are constants for this elsewhere

Reply:
> many things going on here but my main concern is the required changes to RFCs. Let's work out a plan on that first before merging this PR.

I would prefer to remove the additional changes required for the new `encryption` keys in didComm (like recipientEncKeys, EncKey and EncKID) and have this pr merged so we can have legacyKMS completely removed. This will unblock people who need a single source of keys.

Reply:
removed duplicates